discovery-infra/junit_log_parser.py

```diff
@@ -43,66 +44,111 @@ class LogEntry:
+            fail_case.append(flaky_test_case)
+
+        return fail_case
+
+    @classmethod
+    def get_failure_cases(cls, log_file_name: Path, suite_name: str) -> List[TestCase]:
+        fail_cases: Dict[str, List[TestCase]] = dict()
+
+        with open(log_file_name) as f:
+            for line in f:
+                values = re.match(LOG_FORMAT, line)
+                if values is None:
+                    continue
+
+                entry = LogEntry(**values.groupdict())
+                if entry.level not in EXPORTED_LOG_LEVELS:
+                    continue
+                if entry.func not in fail_cases:
+                    fail_cases[entry.func] = list()
+                fail_cases[entry.func] += cls.get_log_entry_case(entry, fail_cases, suite_name)
+
+        log.info(f"Found {len(fail_cases)} failures on {suite_name} suite")
+        return [c for cases in fail_cases.values() for c in cases]
+
+    @classmethod
+    def export_service_logs_to_junit_suites(cls, source_dir: Path, report_dir: Path):
+        suites = list()
+        for file in source_dir.glob("k8s_assisted-service*.log"):
+            suite_name = Path(file).stem.replace("k8s_", "")
+            log.info(f"Creating test suite from {suite_name}.log")
+            test_cases = cls.get_failure_cases(file, suite_name)
+            timestamp = test_cases[0].timestamp if test_cases else None
+            suites.append(TestSuite(name=suite_name, test_cases=test_cases, timestamp=timestamp))
+
+        log.info(f"Generating xml file for {len(suites)} suites")
+        xml_report = to_xml_report_string(suites)
+        with open(report_dir.joinpath(f"junit_log_parser_{str(uuid.uuid4())[:8]}.xml"), "w") as f:
+            log.info(f"Exporting {len(suites)} suites xml-report with {len(xml_report)} characters to {f.name}")
+            f.write(xml_report)
+
+
+class EventsConverter:
+    @classmethod
+    def get_event_test_cases(cls, events_data: dict) -> List[TestCase]:
+        test_cases = list()
+        for event in events_data["items"]:
+            test_case = cls.get_event_test_case(event)
+            test_cases.append(test_case) if test_case else None
+
+        return test_cases
+
+    @staticmethod
+    def get_event_test_case(event: dict) -> Optional[TestCase]:
+        event_type = event["type"]
+        if event_type.lower() in EXPORTED_EVENT_LEVELS:
+            log.info(f"Adding {event_type} event as test-case")
+            reason = event["reason"]
+            involved_object = event["involvedObject"]
+            obj = involved_object["kind"].lower() + "/" + involved_object["name"]
+            message = event["message"]
+            timestamp = event["firstTimestamp"]
+            test_case = TestCase(name=f"{reason}: {obj}", classname="EVENT", category=reason, timestamp=timestamp)
+            test_case.failures.append(CaseFailure(message=message, output=message, type=event_type))
+            return test_case
```

Comment:
nit: add blank line

Reply:
or rather have the ``if event_type.lower() in EXPORTED_EVENT_LEVELS`` short-circuited